src/ol-maplibre-layer.ts

```diff
@@ -6,24 +6,93 @@ import {Map as MaplibreMap} from 'maplibre-gl';
 import type {FrameState} from 'ol/Map.js';
 import type {Options as LayerOptions} from 'ol/layer/Layer.js';
 import type {MapOptions} from 'maplibre-gl';
+import {EventsKey} from 'ol/events';
+import BaseEvent from 'ol/events/Event';
+import MaplibreLayerRenderer from './MaplibreLayerRenderer';
+import {unByKey} from 'ol/Observable';
+import {Source} from 'ol/source';
+import getMaplibreAttributions from './getMaplibreAttributions';
+
+export type MapLibreOptions = Omit<MapOptions, 'container'>;
 
 export type MapLibreLayerOptions = LayerOptions & {
-  maplibreOptions: Omit<MapOptions, 'container'>;
+  maplibreOptions: MapLibreOptions;
+  queryRenderedFeaturesOptions?: maplibregl.QueryRenderedFeaturesOptions;
 };
 
 export default class MapLibreLayer extends Layer {
-  maplibreMap: MaplibreMap;
+  maplibreMap?: MaplibreMap;
+
+  loaded: boolean = false;
+
+  private olListenersKeys: EventsKey[] = [];
 
   constructor(options: MapLibreLayerOptions) {
-    const {maplibreOptions, ...baseOptions} = options;
+    super({
+      source: new Source({
+        attributions: () => {
+          return getMaplibreAttributions(this.maplibreMap);
+        },
+      }),
+      ...options,
+    });
+  }
 
-    super(baseOptions);
+  override disposeInternal() {
+    unByKey(this.olListenersKeys);
+    this.loaded = false;
+    if (this.maplibreMap) {
+      // Some asynchrone repaints are triggered even if the maplibreMap has been removed,
+      // to avoid display of errors we set an empty function.
+      this.maplibreMap.triggerRepaint = () => {};
+      this.maplibreMap.remove();
+    }
+    super.disposeInternal();
+  }
+
+  // @ts-expect-error - this is a mixin
```

Comment:
Again, please remove useless comments, (this one, @private, eslint one, etc.)

Reply:
Done